Ensure JMX is started on XBeanBrokerService

mtaylor opened the pull request:
Using the XBeanBrokerService with JMX and Scheduler Support enabled.
The ManagementContext is never started when in slave mode, since
afterPropertiesSet() blocks trying to get the scheduler shared file
lock.  This patch updates the XBeanBrokerService to start the
BrokerService ManagementContext (when isJMX==true) before trying
to get the job scheduler shared file lock..

gtully commented:
thanks for the contribution, great to see a test :-) however I removed the mods to afterPropertiesSet and the test still works!
One other thought  - the call to startManagementContext needs to be gated with shouldAutostart,
there are use cases where a spring context creates the broker bean for later start up.: 

if (shouldAutostart() && isUseJmx() && !getManagementContext().isStarted()) {
      startManagementContext();
}

mtaylor commented:
@gtully Somehow I removed the key statement from the test when I was cleaning up:

```java
  broker.setSchedulerSupport(true); 
```

It is actually blocking during jobScheduler.start(), whilst the scheduler is trying to get hold of shared file lock.

SharedFileLocker.doStart() line 52:

```java
  while ((!isStopped()) && (!isStopping())) {
```

I'll add back in this statement and make the suggested change.  I wonder though, whether I've added the fix in the correct place?
